modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelect.scala

```diff
@@ -95,11 +103,25 @@ object MajorityPeerSelect {
       )
 
     def getPeerSublist(peers: Set[Peer]): F[List[Peer]] = {
-      val sampleSize = (peers.size * peerSampleRatio).toInt
+      val sampleSize = Math.max((peers.size * peerSampleRatio).toInt, minSampleSize)
 
-      Random[F]
-        .shuffleList(peers.toList)
-        .map(_.take(Math.max(sampleSize, minSampleSize)))
+      for {
+        scores <- getTrustScores.map(_.scores)
+        refinedScores = scores.view
+          .mapValues(score => refineV[TrustValueRefinement](score))
+          .collect {
+            case (key, Right(s)) =>
+              key -> s
+          }
+          .toMap
+        candidates = peers.map { p =>
+          p -> refinedScores.getOrElse(p.id, defaultPeerTrustScore)
+        }.toMap
+        size <- MonadThrow[F].fromEither(
+          refineV[Positive](sampleSize).leftMap(new IllegalStateException(_))
+        )
+        samples <- WeightedProspectSample.sample(candidates, size)
+      } yield samples
```

Comment:
I think it's time to use strategies for peer selection. RandomPeerSelect would be the first strategy, MajorityPeerSelect or TrustBasedPeerSelect would be another one.

Reply:
There will be only one `PeerSelect` that will be used, so there is no need for different strategies. `MajorityPeerSelect` has been renamed `PeerSelect`.

Reply:
Yes, but you already implement a second one (but you remove the previous one). Are we 100% sure we would never use the random peer select again? Making it changeable (strategy design pattern - you have a single trait and multiple implementations) is basically for free, since you already have a trait and one implementation. Instead of removing "RandomBasedPeerSelect", add "TrustBasedPeerSelect" and that's all.

Reply:
Per my discussion with @ryle-goehausen , no, we won't be switching between `PeerSelect` implementations. We will be iterating and upgrading it only.